plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java

```diff
@@ -45,13 +51,28 @@ protected void setup(Binder binder)
     @Singleton
     public static NessieIcebergClient createNessieIcebergClient(IcebergNessieCatalogConfig icebergNessieCatalogConfig)
     {
-        return new NessieIcebergClient(
-                HttpClientBuilder.builder()
-                        .withUri(icebergNessieCatalogConfig.getServerUri())
-                        .withEnableApiCompatibilityCheck(false)
-                        .build(NessieApiV1.class),
+        HttpClientBuilder builder = HttpClientBuilder.builder().fromConfig(getConfigMap(icebergNessieCatalogConfig)::get)
+                .withEnableApiCompatibilityCheck(false);
+
+        return new NessieIcebergClient(builder.build(NessieApiV1.class),
                 icebergNessieCatalogConfig.getDefaultReferenceName(),
                 null,
                 ImmutableMap.of());
     }
+
+    private static Map<String, String> getConfigMap(IcebergNessieCatalogConfig catalogConfig)
+    {
+        Map<String, String> configMap = new HashMap<>();
+        configMap.put(NessieConfigConstants.CONF_NESSIE_REF, catalogConfig.getDefaultReferenceName());
+        URI serverUri = catalogConfig.getServerUri();
+        if (serverUri != null) {
```

Comment:
In which context would it make sense to have the server not present?

Reply:
when user forget configure? We don't have a default value for the uri. 
Since we are doing a toString(), I am checking for the null to avoid NPE.

Reply:
removed the null check.